### DISCUSSION TITLE
Add the ability to "Reset" a certificate.

### DISCUSSION THREAD

**Discussion by ferneaux** [Venafi/venafips#35] [Ideas]
I've been unable to locate a cmdlet or switch that let's me reset the status/stage of a certificate. For example, if there is a problem renewing or deploying a cert, you can't renew it until the status is reset/cleared.

Is it possible to have a cmdlet like "Reset-TppCertificateStatus" that uses the cert DN path or can be used as part of a pipeline?

That would save a lot of clicking through the web ui.

Thanks!

**Comment by gdbarron** [comment]
I believe this is what you're looking for?

https://docs.venafi.com/Docs/21.3SDK/TopNav/Content/SDK/WebSDK/r-SDK-POST-Certificates-Reset.php

**Reply by ferneaux** [reply]
That looks like it'll do the trick... thanks!

I just plug in the needed API calls into the "Invoke-VenafiRestMethod" cmdlet and that should be the end of that, correct? (well, in theory anyways)

**Reply by gdbarron** [reply]
`Invoke-VenafiRestMethod -Method Post -UriLeaf 'Certificates/Reset' -VenafiSession $VenafiSession -Body @{'CertificateDN'='\ved\policy\mycert.com'}` should get it done.

If you think this will be useful going forward I'm happy to add it to the module.  If you could add this as an enhancement issue that would be great!

**Reply by ferneaux** [reply]
Worked like a charm! Thank you!

**Reply by gdbarron** [reply]
@ferneaux check out `Invoke-VenafiCertificateAction` released in v3.4.  All the standard actions are there including reset.

**Reply by ferneaux** [reply]
Yeah... installed it last week and it is a LIFE SAVER!!! Thanks x1 million!!
